sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogV2Util.scala

```diff
@@ -296,6 +297,49 @@ private[sql] object CatalogV2Util {
     }
   }
 
+  /**
+   * Apply Table Constraints changes to an existing table and return the result.
+   */
+  def applyConstraintChanges(
+      table: Table,
+      changes: Seq[TableChange]): Array[Constraint] = {
+    val constraints = table.constraints()
+
+    def findExistingConstraint(name: String): Option[Constraint] = {
+      constraints.find(_.name.toLowerCase(Locale.ROOT) == name.toLowerCase(Locale.ROOT))
+    }
+
+    changes.foldLeft(constraints) { (constraints, change) =>
+      change match {
+        case add: AddConstraint =>
+          val newConstraint = add.constraint
+          val existingConstraint = findExistingConstraint(newConstraint.name)
+          if (existingConstraint.isDefined) {
+            throw new AnalysisException(
+              errorClass = "CONSTRAINT_ALREADY_EXISTS",
+              messageParameters =
+                Map("constraintName" -> existingConstraint.get.name,
+                  "oldConstraint" -> existingConstraint.get.toDDL))
```

Comment:
do we really need to display the old constraints? Are there security issues?

Reply:
It would be easier for users to know what is in the old constraint without running a describe table command and find details.
Constraints should be simple. We will show all the constraints in describe table command anyway.